### DISCUSSION TITLE
[Feature Request] Option to Choose Container Startup After Creation

### DISCUSSION THREAD

**Discussion by mschabhuettl** [tteck/Proxmox#1471] [Ideas]
Hello,

I would like to propose adding a feature to the Proxmox VE Helper Scripts. It would be great if there was a question in the script that asks the user if they want to start the container immediately after its creation. This feature is similar to the HAOS VM functionality and would enhance the user experience by providing the choice to start or not start the container, as I don't want the container to be started automatically.

Thank you for considering this suggestion.

Best regards,
Matthias Schabhüttl

**Comment by tteck** [comment]
In contrast to a virtual machine, a container needs to be started after its creation in order to install the application and its accompanying dependencies.

**Reply by mschabhuettl** [reply]
Is it possible to add a delay?

**Reply by tteck** [reply]
May I ask why?

**Reply by mschabhuettl** [reply]
Sure, I utilize DHCP Reservation in my workflow. To make it work, I made a modification to your `build.func` script by adding a `sleep 120` command to introduce a 120-second delay before the LXC starts. Here are the steps I follow:

1. Modify your script to include a 120-second delay before starting execution.
2. Launch my modified version of the install-script.
3. Proceed with the necessary configurations, settings, etc. of the LXC
4. Before the 120-seconds are up, I take note of the Randomized MAC address of the Network Interface Card (NIC) and set up a DHCP reservation for the desired IP address.

**Reply by mschabhuettl** [reply]
@tteck My suggestion would be to add a "Delay startup by X seconds"-Input (Default = 0s) as an advanced option.

**Reply by tteck** [reply]
Why not configure the preferred IP address through static IP settings in the Advanced settings?

**Reply by tteck** [reply]
Alternatively, you can create your own MAC address in advance, specify it for DHCP reservation, and then enter the MAC into the Advanced settings. 
```
MAC=02:$(openssl rand -hex 5 | awk '{print toupper($0)}' | sed 's/\(..\)/\1:/g; s/.$//')
echo "$MAC"
```

**Reply by mschabhuettl** [reply]
I'm aware of that. And static IP would be an option, but I just like DHCP Reservation much more.

**Reply by AlexGeek77** [reply]
@tteck is giving you a way to have a DHCP reservation. For me, I have a sequential list of MAC addresses that I use for all my homelab servers, then during the install, on the advanced section, I type in one of my MAC addresses and then create the reservation before continuing on to the next step in the script. Works like a charm.
